### PR TITLE
Add logic to delete songs and send a welcome email

### PR DESCRIPTION
**Pull request by PashkovichKaterina**
**Task:** Implement the ability to delete a song and send a welcome email after registration.

**Solution steps:**
1. Update `DATETIME_FORMAT` to iso-8601.
2. Implement the ability to delete song file from the AWS S3 bucket.
3. Add test for song delete method.
4. Implement a welcome email after registration using Sendinblue.
5. Create GitHub Action to deploy backend automatically to AWS after the pull request is closed.

### CODE REVIEW

File: backend/settings.py [hunk 1]
```diff
@@ -185,4 +181,13 @@
     }
 }
 
-DATETIME_FORMAT = "%b %d %Y %H:%M"
+DATETIME_FORMAT = "iso-8601"
+
+EMAIL_BACKEND = "anymail.backends.sendinblue.EmailBackend"
+ANYMAIL = {
+    "SENDINBLUE_API_KEY": os.environ["SENDINBLUE_API_KEY"]
```
Comment (artsiom-streetshares) [inline backend/settings.py:188]: good that you store it in env. but this is a secret, so best practice is to either keep it encrypted, and then decrypt in runtime, or use secret storage like https://aws.amazon.com/secrets-manager/

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - .env
+
+  nginx:
+    build: ./nginx
+    ports:
+      - "80:80"
+    depends_on:
+      - app
+
+  redis:
```
Comment (artsiom-streetshares) [inline docker-compose.yaml:24]: so you deploy redis as a part of your elastic beanstalk environment, correct?

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+version: "3"
+
+services:
+  app:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: >
+    sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
```
Comment (artsiom-streetshares) [inline docker-compose.yaml:9]: let's change django server to gunicorn at some point

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    volumes:
+      - .:/app
+    expose:
+      - "8000"
+    env_file:
+      - .env
+
+  nginx:
```
Comment (artsiom-streetshares) [inline docker-compose.yaml:17]: what is nginx for in your setup?

**Review by artsiom-streetshares — APPROVED** [review]

File: backend/settings.py (hunk 1 above)
Comment (PashkovichKaterina) [reply backend/settings.py]: I used AWS Secrets Manager for this:
1. I created a new policy to grant access to secrets and attached it to the Elastic Beanstalk instance role.
2. Made changes to the code.

File: docker-compose.yaml (hunk 2 above)
Comment (PashkovichKaterina) [reply docker-compose.yaml:23]: Yes, Redis runs in a docker container on Elastic Beanstalk instance.

File: docker-compose.yaml (hunk 4 above)
Comment (PashkovichKaterina) [reply docker-compose.yaml:16]: I used to use `Dockerfile` to deploy my project to Elastic Beanstalk and it worked. But when I added `docker-compose.yaml` I got a 502 error. I added nginx setting to solve this problem.

File: docker-compose.yaml (hunk 2 above)
Comment (artsiom-streetshares) [reply docker-compose.yaml:23]: this is good. though this setup might be problematic if you'll need to scale. as alternative take a look at AWS ElastiCache service, it's managed redis in AWS

File: docker-compose.yaml (hunk 4 above)
Comment (artsiom-streetshares) [reply docker-compose.yaml:16]: maybe it was because of server wasn't mapped to port 80. but your setup is ok

**Pull request merged**